### ISSUE TITLE
Ensure all functions return a consistently typed and formatted object (eg `getCohortGenerationInformation()` )

### ISSUE THREAD

**Issue by ablack3** [OHDSI/ROhdsiWebApi#168]
`getCohortGenerationInformation()` returns a tibble with with zero columns when no generation information is found but with 13 columns when cohort generation information is found. This incosistancy in the format of the return object might cause unexpected errors when used in other packages. I propose that `getCohortGenerationInformation()` should always return a tibble with 13 columns. If no generations are found then the tibble will have 0 rows. More generally all ROhdsiWebApi functions should return a consistently typed/formatted object or throw an error.

**Comment by gowthamrao** [comment]
@ablack3  is this still an active issue?

**Comment by ablack3** [comment]
Yes this is still a problem. I don't think ROhdsiWebApi functions should ever return a 0x0 tibble. The reason is that if I'm retrieving data for many cohorts and want to combine the results I would like to row bind all of the resulting dataframes together.  I think we should aim to create functions that always return a constantly formatted object (e.g. a dataframe with prespecified columns) over functions that are not type consistent (e.g. sapply ).


``` r
library(ROhdsiWebApi)
getCohortGenerationInformation(1776538, "http://atlas-demo.ohdsi.org/WebAPI")
#> Found 0 generations for Cohort of which NA had a status = 'COMPLETED'
#> # A tibble: 0 x 0

# Expected/desired result
# A tibble: 0 x 11
# ... with 11 variables: cohortDefinitionId <int>, sourceId <int>, startTime <dttm>, executionDuration <int>,
#   status <chr>, isValid <lgl>, isCanceled <lgl>, personCount <int>, recordCount <int>, sourceKey <chr>,
#   sourceName <chr>
```

<sup>Created on 2021-03-04 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

**Comment by gowthamrao** [comment]
@ablack3  do you want to take this on and submit a PR . we can target it to version 1.2 release

**Comment by gowthamrao** [comment]
btw - dplyr::bind_rows function can bind tibble 0 x 0 with tibble 0 x 11

**Comment by ablack3** [comment]
Yes I’ll give it a shot.

**Comment by gowthamrao** [comment]
@ablack3 where are we on this one? Can we move this to Release 1.3?

**Comment by ablack3** [comment]
I have not had a chance to spend time on it. Yes let's move it to 1.3 or you can close the issue and we can wait to see if any new issues are caused by 0 column dataframes.

**Comment by gowthamrao** [comment]
@ablack3 have you had a chance to work on this yet?

**Comment by ablack3** [comment]
No I have not had a chance to make any changes. Making a note that the "exist" functions are type inconsistent. https://github.com/OHDSI/ROhdsiWebApi/blob/db363a4e5d77dfa1a68267a9e1e4697ec46c02f1/R/AutoGeneratedDefinitions.R#L123
These might return a dataframe or a boolean value. 
I would like to change these so that they always return a boolean value or always return a dataframe. This would be part of a major release since it is breaking change.
